plasmapy/utils/pytest_helpers.py

```diff
@@ -719,3 +720,145 @@ def run_test_equivalent_calls(*test_inputs, require_same_type: bool = True):
                 f"of type {test_case['type']}")
 
         raise UnexpectedResultError(errmsg)
+
+
+def assert_can_handle_nparray(function_to_test, insert_some_nans=[], insert_all_nans=[],
+                              kwargs={}):
+    """
+    Test for ability to handle numpy array quantities.
+
+    Parameters
+    ----------
+    function_to_test
+        The function to be tested for ability to handle numpy array quantities.
+        Arguments are automatically given a vector input based on their
+        variable name. Current args that are interpreted as vectors are:
+        `["T", "T_i", "T_e", "temperature"]`
+        `["n", "n_i", "n_e", "density"]`
+        `["B"]`
+        `["V", "Vperp"]`
+        `["coulomb_log"]`
+        `["characteristic_length"]`
+
+    insert_some_nans: list
+        List of argument names in which to insert some np.nan values.
+        These must be arguments that will be tested as vectors as listed
+        above.
+
+    insert_all_nans: list
+        List of argument names to fill entirely with np.nan values.
+
+    kwargs: dict
+        Arguments to pass directly to the function in under test, in the
+        normal kwargs python dictionary format.
+
+    Raises
+    ------
+    ~ValueError
+        If this function cannot interpret a parameter of function_to_test,
+
+    Examples
+    --------
+    >>> from plasmapy.physics.parameters import Alfven_speed, gyrofrequency
+    >>> assert_can_handle_nparray(Alfven_speed)
+    >>> assert_can_handle_nparray(gyrofrequency, kwargs={"signed": True})
+    >>> assert_can_handle_nparray(gyrofrequency, kwargs={"signed": False})
+    """
+    def _prepare_input(param_name, param_default, insert_some_nans, insert_all_nans, kwargs):
+        """
+        Parse parameter names and set up values to input for 0d, 1d, and 2d array tests.
+        """
+        # first things first: let any passed in kwarg right through (VIP access)
+        if param_name in kwargs.keys():
+            return kwargs[param_name], kwargs[param_name], kwargs[param_name]
+
+        # else, if it's a recognized variable name, give it a reasonable unit and magnitude
+        elif param_name in ["particle", "ion_particle", "ion"]:
+            if not (param_default is inspect._empty or param_default is None):
+                return param_default, param_default, param_default
+            else:
+                return "p", "p", "p"
+        elif param_name == "particles":
+            if not (param_default is inspect._empty):
+                return param_default, param_default, param_default
+            else:
+                return ("e", "p"), ("e", "p"), ("e", "p")
+        elif param_name in ["T", "T_i", "T_e", "temperature"]:
+            unit = u.eV
+            magnitude = 1.0
+        elif param_name in ["n", "n_i", "n_e", "density"]:
+            unit = u.m ** -3
+            magnitude = 1e20
+        elif param_name == "B":
+            unit = u.G
+            magnitude = 1e3
```

Comment:
Sure, but I thought we were going with reasonable units? :laughing: I kid. It's the first time I'm seing good old Gauss in `astropy.units`, though.

Reply:
Haha, oh you don't like Gauss?  You should try it out, way better than Tesla. :smiling_imp: Especially kilogauss and megagauss.